redis/src/main/scala/zio/redis/commands/Cluster.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2021 John A. De Goes and the ZIO contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.redis.commands
+
+import zio.Chunk
+import zio.redis.Input._
+import zio.redis.Output._
+import zio.redis._
+import zio.schema.codec.BinaryCodec
+
+private[redis] trait Cluster extends RedisEnvironment {
+  import Cluster._
+
+  final val _asking: RedisCommand[Unit, Unit] = askingCommand(codec, executor)
```

Comment:
Two things:
- let's not name things like this
- are these exist to "help" with `transactional` package?

Generally speaking, they may be useful, but we don't have a "proof" for that (i.e. benchmarks), and they don't improve anything on schema-dependent commands.

Reply:
These are used by both regular API and Transactional one. The idea was to define a command once and use it from both APIs. It was not introduced for optimization purposes. Also, I've done this because I don't know how to use the same API for both. The reason is the runtime behaviour. Regular API sends a command and receives a response, while the transactional one sends a command and has to remember what type of output it would have to parse later when `EXEC` is sent. That's why I had to introduce `RedisTransaction` which is basically transactional command.